sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2350,6 +2350,13 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
+  val ANSI_INTERVALS_ENABLED = buildConf("spark.sql.ansi.intervals.enabled")
```

Comment:
I think the question is if we should use the new interval types by default or not. If not, I'd prefer to use the ansi flag directly. If yes, I'd prefer to create a new legacy config and disable the legacy behavior by default.

Reply:
+1, Looks like only spark have type  `CalendarIntervalType`. In other engine like hive, presto etc, they use day time interval and year month interval too.

Reply:
The question is more complex because there is no feature parity of new intervals with the existing one. We cannot disable `CalendarIntervalType` completely at the moment. Also the legacy config means that we cannot expose it to users and mention in our public docs. But I guess we will have to mention it in docs.